### ISSUE TITLE
Combobox: problems when closing popup

### ISSUE THREAD

**Issue by mythic2** [ibm-js/deliteful#582]
I do not know whether to call it a "race condition" problem as the result is always the same for me. But I believe it is not intended workflow.

When the uses selects something in the combobox, the dropdown should be closed:
1. HasDropDown.closeDropDown is called. The function contains following code 

``` javascript
...
if (this.opened) {
    this.popupStateNode.setAttribute("aria-expanded", "false");
    if (focus && this.focus) {
        this.focus();
    }

    this.emit("delite-before-hide", {
        child: this._currentDropDown,
        cancelable: false
    });

    popup.close(this._currentDropDown);
    this.opened = false;
...
```
1. the `this.focus()` call cause change of the selection stack, which causes call to `closeDropDown` again. The popup is still marked as opened so the method performs all actions and in the end it calls `delete this._currentDropDown`
2. `this.focus()` returns and the original closeDropDown continues, however `this._currentDropDown` is undefined now. Which causes two problems for me:
   - `popup.close(undefined)` closes all popups, making it impossible to have a comobox in a popup frame
   - `this._currentDropDown.style.cssText` throws an exception

See the stack trace demonstrating that closeDropDown is called recursively twice.
dcl.closeDropDown (HasDropDown.js:604)
register.closeDropDown (Combobox.js:761)
dcl._deactivatedHandler (HasDropDown.js:407)
dcl.emit (CustomElement.js:422)
dcl._setStack (activationTracker.js:272)
dcl._pointerDownOrFocusHandler (activationTracker.js:208)
dcl._focusHandler (activationTracker.js:246)
focusHandler (activationTracker.js:102)
dcl.focus (FormWidget.js:152)
dcl.closeDropDown (HasDropDown.js:596)
register.closeDropDown (Combobox.js:761)
(anonymous function) (Combobox.js:419)
(anonymous function) (Destroyable.js:108)
setTimeout (async)
dcl.defer (Destroyable.js:111)
(anonymous function) (Combobox.js:416)

The code is based on master versions of delite and deliteful running in Chrome 44

**Comment by wkeese** [comment]
Hmm, sounds like a HasDropDown bug, I'll take a look.  But does this happen in our unit/functional tests?  It seems like something we would have caught already.  Perhaps the exceptions occur but the code still keeps running fine.

**Comment by wkeese** [comment]
The problem you described won't happen anymore thanks to 67b28f9d978f010dd0776d1dbf17a58d140e77f0, which saves `this._currentDropDown` in a local variable.

I'd still be concerned about the double call to `closeDropDown()` that you described, but I'm not seeing that either [anymore], at least not on the latest code on Chrome/mac.  I figured breakpoints might change the behavior, so instead I tested with `console.log()` statements, and I'm only seeing one call to `closeDropDown()`.
